Content.Server/Chat/Systems/ChatSystem.cs

```diff
@@ -181,6 +183,9 @@ public void TrySendInGameICMessage(
             return;
         }
 
+        if (TryComp<CollectiveMindComponent>(source, out var collective))
+            _collectiveMind.UpdateCollectiveMind(source, collective);
```

Comment:
This does expensive `O(n^2)` proto enumeration, component+tag enumeration on _every_ chat message sent.